tests/unit/DebugUtilsTest.ts

```diff
@@ -0,0 +1,546 @@
+            expect(DebugUtils.stringToOnyxData('false', 'boolean')).toBe(false);
+        });
+        it('returns null when type is undefined', () => {
+            expect(DebugUtils.stringToOnyxData('2', 'undefined')).toBe(null);
+        });
+        it('returns string when type is string', () => {
+            expect(DebugUtils.stringToOnyxData('2', 'string')).toBe('2');
+        });
+        it('returns string when type is symbol', () => {
+            expect(DebugUtils.stringToOnyxData('2', 'symbol')).toBe('2');
+        });
+        it('returns string when type is function', () => {
+            expect(DebugUtils.stringToOnyxData('2', 'symbol')).toBe('2');
+        });
+        it('returns string when type is not specified', () => {
+            expect(DebugUtils.stringToOnyxData('2')).toBe('2');
+        });
+    });
+    describe('compareStringWithOnyxData', () => {
+        it('returns true when data is undefined and text is "undefined"', () => {
+            expect(DebugUtils.compareStringWithOnyxData('undefined', undefined)).toBe(true);
+        });
+        it('returns false when data is undefined and text is not "undefined"', () => {
+            expect(DebugUtils.compareStringWithOnyxData('A', undefined)).toBe(false);
+        });
+        it('returns true when data is object and text is data in JSON format', () => {
+            expect(DebugUtils.compareStringWithOnyxData(MOCK_DRAFT_REPORT_ACTION, MOCK_REPORT_ACTION)).toBe(true);
+        });
+        it('returns false when data is object and text is not data in JSON format', () => {
+            expect(DebugUtils.compareStringWithOnyxData('{}', MOCK_REPORT_ACTION)).toBe(false);
+        });
+        it('returns true when data is string and text equals to data', () => {
+            expect(DebugUtils.compareStringWithOnyxData('A', 'A')).toBe(true);
+        });
+        it('returns false when data is string and text is not equal to data', () => {
+            expect(DebugUtils.compareStringWithOnyxData('2', 'A')).toBe(false);
+        });
+    });
+    describe('getNumberOfLinesFromString', () => {
+        it('returns 1 when string is empty', () => {
+            expect(DebugUtils.getNumberOfLinesFromString('')).toBe(1);
+        });
+        it('returns 1 when no "\\n" are present in the string', () => {
+            expect(DebugUtils.getNumberOfLinesFromString('Something something something')).toBe(1);
+        });
+        it('returns k when there are k - 1 "\\n" present in the string', () => {
+            expect(DebugUtils.getNumberOfLinesFromString('Line1\n Line2\nLine3')).toBe(3);
+        });
+    });
+    describe('onyxDataToDraftData', () => {
+        it('returns an empty object if data is an empty object', () => {
+            expect(DebugUtils.onyxDataToDraftData({})).toEqual({});
+        });
+        it('returns a converted object where all values are string representations of the original values from data object', () => {
```

Comment:
Not sure if this test is really valid -- you are basically testing if `JSON.stringify()` is doing its job correctly, but we shouldn't do unit tests for core JS functions